agent/engine/docker_task_engine.go

```diff
@@ -597,6 +612,15 @@ func (engine *DockerTaskEngine) createContainer(task *api.Task, container *api.C
 	seelog.Infof("Created container name mapping for task %s - %s -> %s", task, container, containerName)
 	engine.saver.ForceSave()
 
+	// Create metadata directory and file then populate it with common metadata of all containers of this task
+	// Afterwards add this directory to the container's mounts if file creation was successful
+	// CreateMetadata has a call to docker API for the client version so this may be block our container creation
+	// But we require the host Config binds to be properly set before we create the container
+	mderr := engine.metadataManager.CreateMetadata(&hostConfig.Binds, task, container)
```

Comment:
I think many people have asked about this and I forgot to respond to this. err is already used above as a different type so using err here as well will cause type conflicts.